lib/dialects/abstract/query-generator.js

```diff
@@ -1906,6 +1907,14 @@ class QueryGenerator {
       fragment += ` AS ${mainTableAs}`;
     }
 
+    if (options.indexHints && this._dialect.supports.indexHints) {
+      for (const hint of options.indexHints) {
+        if (IndexHints[hint.type]) {
+          fragment += ` ${IndexHints[hint.type]} INDEX (${hint.values.join(',')})`;
```

Comment:
You will need to quote index names from `values`

```js
hint.values.map(indexName => this.quoteIdentifiers(indexName)).join(',')
```

Reply:
Updated in 65c91b178d1f5427489ac458298cd2557d2c135e